### PR TITLE
Add prettier action to pull requests

### PR DESCRIPTION
**Pull request by noahp**
### Summary

From https://github.com/memfault/memfault-docs .

Only files changed in the pull request will be formatted, see
[`.github/workflows/lint.yml`](.github/workflows/lint.yml).

Pull requests from branches on this repo will have an automatic commit applied
with the formatting. Pull requests from forks will **fail** CI, but not have a
commit applied.

 ### Test Plan

Test by touching a file under the following circumstances:

  - [x] the pull request is from a branch on this repo
    - https://github.com/memfault/interrupt/pull/243
    - the "Prettified code!" commit is auto-pushed to the branch
  - [x] the pull request is from a fork
    - https://github.com/memfault/interrupt/pull/244
    - the PR check fails: https://github.com/memfault/interrupt/runs/4938847953?check_suite_focus=true#step:5:19

### CODE REVIEW

File: .prettierrc.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+overrides:
+  - files: "*.md"
+    options:
+      proseWrap: always
```
Comment (tyhoff) [inline .prettierrc.yml:4]: Are we sure we want this? It frequently mangles quote blocks and blocks of links. 

Have you had experiences with this?

**Comment by tyhoff** [comment]
As much as I want this, I don't think it's ready for prime time. As a first example, it broke the lists in the IDAP post (first one ever!)

image
image

File: .prettierrc.yml (hunk 1 above)
Comment (noahp) [reply .prettierrc.yml:4]: My experience has been pretty good with wrapping enabled, but it requires marking code blocks / quote blocks correctly :(

I'll change it to run only on changed files, and only on PRs (so we don't get a surprise on master).

**Comment by noahp** [comment]
> As much as I want this, I don't think it's ready for prime time. As a first example, it broke the lists in the IDAP post (first one ever!)
> ...
> Or maybe we can comb through those issues. That issue above stems from the fact that the code block was not indented.

Very good call, I've updated it to only run on changed files and ONLY on PRs (so we don't get surprise commits). One thing to consider:

- currently this will auto-push to branches from this repo, and fail CI on PRs with branches from forked repos. this might be a bad/frictiony experience for users that have difficulty running `prettier`

**Comment by noahp** [comment]
Closing, it's too much effort to get the skip behavior I want. The simple support github provides also disables the netlify preview action :confused: not ideal.

https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

**Pull request closed without merging**